include/deal.II/algorithms/any_data.h

```diff
@@ -231,10 +232,10 @@ template <typename type>
 inline type
 AnyData::entry(const unsigned int i)
 {
-  AssertIndexRange(i, size());
+  DEAL_II_AssertIndexRange(i, size());
   type *p = boost::any_cast<type>(&data[i]);
-  Assert(p != nullptr,
-         ExcTypeMismatch(typeid(type).name(), data[i].type().name()));
+  DEAL_II_Assert(p != nullptr,
```

Comment:
I know that this will create a crazy amount of bikeshedding - but I am not happy with the mix of camel case and all caps.
What about one of the variants below?
```
DEAL_II_ASSERT with DEAL_II_ASSERT_INDEX_RANGE
DealAssert with DealAssertIndexRange
deal_assert with deal_assert_index_range
```

Reply:
and honestly, what about we just bite the bullet and keep `Assert` and `AssertIndexRange`?

Reply:
At least `Assert` is incompatible with the `gtest/gmock` framework, according to @luca-heltai in https://github.com/dealii/dealii/issues/7306#issuecomment-430790018.

Reply:
I created a poll in `Slack`.